Source/Spray/SprayParticles.cpp

```diff
@@ -327,10 +327,11 @@ SprayParticleContainer::updateParticles(
         refv.fillPtrs_d(rf_d);
       }
       auto* N_SB = N_SB_d.dataPtr();
+      auto const* leosparm = m_sprayData->eosparm;
       amrex::ParallelFor(Np, [=] AMREX_GPU_DEVICE(int pid) noexcept {
         ParticleType& p = pstruct[pid];
         if (p.id() > 0) {
-          auto eos = pele::physics::PhysicsType::eos();
+          auto eos = pele::physics::PhysicsType::eos(leosparm);
```

Comment:
use `fdat->eosparm`. This code will give you the host version, but this call is within the parallel for so you need the device version. Then you also won't need the line above defining leosparm.